File: code/intermediate/polymorphism/covariantReturnType/CovariantReturnType.js
function A() { }
A.prototype.get = function () {
    return this; //this refers to A constructor
}

function B() { }

B.prototype = Object.create(A.prototype);
B.prototype.constructor = B;
B.prototype.get = function () {
    return this; //this refers to B constructor
}

B.prototype.getMessage = function () {
    console.log("welcome to covariant return type");
}

const b = new B();
b.get().getMessage();

File: code/intermediate/polymorphism/covariantReturnType/CovariantReturnType.java
class A {
    A get() {
        return this;
    }
}

class B {
    B get() {
        return this;
    }

    void getMessage() {
        System.out.println("Welcome covariant type");
    }
    public static void main(String args[]){
        new B().get().getMessage();
    }
}